pkg/models/protocolvalidator.go

```diff
@@ -0,0 +1,16 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package models
+
+import "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+// ProtocolValidator is a low-level device-specific interface implemented
+// by device services that validate device's protocol properties.
+type ProtocolValidator interface {
```

Comment:
Can we call this DeviceValidator and the protocol validation is part of this validation?
```suggestion
type DeviceValidator interface {
```

File name would change also